packages/headless/src/api/platform-client.ts

```diff
@@ -109,7 +109,19 @@ export class PlatformClient {
         };
       }
 
-      throw error;
+      if (error.name === 'AbortError') {
+        throw error;
+      }
+
+      // Transform an error to an object https://stackoverflow.com/a/26199752
+      const errorObj = JSON.parse(
```

Comment:
Those "unknown" errors are mostly... client side error, unless I'm mistaken ?

Like, your internet disconnected, you have a firewall rule blocking coveo platform, things like that.

Because all of the others that would "platform related" would come back from the API with an error response, right ?

Open question: Would it be better to return an object where we control the body/message with something like "client side error" + the generic error.message.

It might help troubleshooting ? If you see that it's a client side error, there's not much that Coveo can do, most probably, to help you fix it.

Reply:
If there is a way to differentiate errors that arise from not being able to contact the server due to a local issue, and not because the api is down (code 5xx), that would be very cool.

Reply:
Yes for all!